cg/meta/archive/archive.py

```diff
@@ -188,7 +186,104 @@ def add_samples_to_files(self, files_to_archive: list[File]) -> list[FileAndSamp
+            self.update_retrieval_jobs_for_archive_location(
+                archive_location=archive_location,
+                job_ids=retrieval_ids_per_location.get(archive_location),
+            )
+
+    def update_archival_jobs_for_archive_location(
+        self, archive_location: ArchiveLocations, job_ids: list[int]
+    ) -> None:
+        for job_id in job_ids:
+            self.update_ongoing_task(
+                task_id=job_id, archive_location=archive_location, is_archival=True
+            )
+
+    def update_retrieval_jobs_for_archive_location(
+        self, archive_location: ArchiveLocations, job_ids: list[int]
+    ) -> None:
+        for job_id in job_ids:
+            self.update_ongoing_task(
+                task_id=job_id, archive_location=archive_location, is_archival=False
+            )
+
+    def update_ongoing_task(
+        self, task_id: int, archive_location: ArchiveLocations, is_archival: bool
+    ) -> None:
+        """Fetches info on an ongoing job and updates the Archive entry in Housekeeper."""
+        archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[archive_location](self.data_flow_config)
+        is_job_done: bool = archive_handler.is_job_done(task_id)
+        if is_job_done:
+            LOG.info(f"Job with id {task_id} has finished, updating Archive entries.")
+            if is_archival:
+                self.housekeeper_api.set_archived_at(task_id)
+            else:
+                self.housekeeper_api.set_retrieved_at(task_id)
+        else:
+            LOG.info(f"Job with id {task_id} has not yet finished.")
+
+    def sort_archival_ids_on_archive_location(
+        self, jobs: list[Archive]
+    ) -> dict[ArchiveLocations, list[int]]:
+        """Returns a dictionary with keys being ArchiveLocations and the values being the subset of the given
+        archival jobs which should be archived there."""
+        handled_job_ids: list[int] = []
+        jobs_per_location: dict[ArchiveLocations, list[int]] = {}
+        for job in jobs:
+            if job.archiving_task_id in handled_job_ids:
+                continue
+
+            archive_location = ArchiveLocations(self.get_archive_location_from_file(job.file))
+            if jobs_per_location.get(archive_location):
+                jobs_per_location[archive_location].append(job.archiving_task_id)
+            else:
+                jobs_per_location[archive_location] = [job.archiving_task_id]
+        return jobs_per_location
+
+    def sort_retrieval_ids_on_archive_location(
+        self, jobs: list[Archive]
+    ) -> dict[ArchiveLocations, list[int]]:
+        """Returns a dictionary with keys being ArchiveLocations and the values being the subset of the given
+        retrieval jobs which should be archived there."""
+        handled_job_ids: list[int] = []
+        jobs_per_location: dict[ArchiveLocations, list[int]] = {}
+        for job in jobs:
+            if job.retrieval_task_id in handled_job_ids:
+                continue
```

Comment:
I don't really understand what `handled_job_ids` does

Reply:
It is not so beautiful but you can have multiple entries in Archive per job_id, and I only wanted the job_id once if that makes sense. Probably better with something like:

```Python
unique_job_ids: set(int) = set([job.retrieval_task_id for job in jobs]
```

Reply:
@Vince-janv Have a look now, tried to refactor it a bit!

Reply:
I think this is the cleanest we can get it. A thought though for a future PR. Would it not make sense to also store the archive location in housekeeper? I might have been against it before but I think it can really simplify our code 🤔 What are your thoughts?

Reply:
I would be in favour! Just as a column in the Archive table maybe? Or if we don't want to duplicate data, maybe just make it into a property?

Reply:
Will refactor in a separate PR.